week7/src/components/LetterForm.tsx

```diff
@@ -0,0 +1,160 @@
+import styled from 'styled-components';
+import { useRef } from 'react';
+import { Letter } from '../types/Letter';
+import { useNavigate } from 'react-router-dom';
+import { client } from '../services';
+
+interface LetterFormProps {
+  letterInfo: Letter;
+}
+
+interface BodyObject {
+  [key: string]: string; //index signature
+}
+
+export default function LetterForm({ letterInfo }: LetterFormProps) {
+  let navigate = useNavigate();
+  const isEditing = letterInfo ? true : false;
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const inputInfo = [
+    { label: '이름', id: 'name', placeholder: '이름이 뭐에요?' },
+    {
+      label: '내용',
+      id: 'content',
+      placeholder: '무슨 내용의 편지를 써볼까요?',
+    },
+    {
+      label: '비밀번호',
+      id: 'password',
+      placeholder: '비밀번호를 통해 편지를 잠궈보아요.',
+      type: 'password',
+    },
+    {
+      label: '비밀번호 힌트',
+      id: 'hint',
+      placeholder: '누군가 내 비밀편지를 보도록 비밀번호 힌트를 주세요.',
+    },
+  ];
+
+  const onSubmitForm = async (e: any) => {
+    e.preventDefault();
```

Comment:
```suggestion
const onSubmitForm = async (e: FormEvent) => {
    e.preventDefault();
    if (!e.target || !(e.target instanceof Array<HTMLInputElement>)) return;
```

요렇게 내로잉하는 방법도 있을 것 같소~

Reply:
오............이런 방법이 있군아............ㅋㅋㅋㅋㅋㅋㅋㅋ 좋다...
고마워!!